project4/src/AI.cpp

```diff
@@ -0,0 +1,311 @@
+        eval += 1;
+    eval += 5 * evaluateVertically(board,xCoordinate,yCoordinate,player);
+    eval += 5 * evaluateHorizontally(board,xCoordinate,yCoordinate,player);
+    eval += 5 * evaluateDiagonally(board,xCoordinate,yCoordinate,player);
+    eval += 5 * evaluateAntidiagonally(board,xCoordinate,yCoordinate,player);
+
+    return eval;
+}
+
+int AI::evaluateVertically(const Board &board, int xCoordinate, int yCoordinate, char currentPlayer) const
+{
+    int start =0, end = board.getSize()-1;
+    if(yCoordinate - board.getWinningNumber() +1 >= 0 ) start = yCoordinate - board.getWinningNumber() +1;
+    if(yCoordinate + board.getWinningNumber() - 1 <= end) end = yCoordinate + board.getWinningNumber() - 1;
+    int counterTop = 0, counterBottom = 0;
+    int opponentCounterTop = 0, opponentCounterBottom = 0;
+    for(int i = start; i < yCoordinate; ++i)
+    {
+        if (board.getFields()[xCoordinate][i]==currentPlayer)
+        {
+            counterBottom++;
+            opponentCounterBottom = 0;
+        }
+        else if (board.getFields()[xCoordinate][i]== ' ');
+        else
+        {
+            opponentCounterBottom++;
+            counterBottom = 0;
+        }
+    }
+
+    for(int i = end; i > yCoordinate; --i)
+    {
+        if (board.getFields()[xCoordinate][i]==currentPlayer)
+        {
+            counterTop++;
+            opponentCounterTop = 0;
+        }
+        else if (board.getFields()[xCoordinate][i]== ' ');
+        else
+        {
+            opponentCounterTop++;
+            counterTop = 0;
+        }
+    }
+    if(opponentCounterBottom+opponentCounterTop == board.getWinningNumber() - 1) return 1000;
+    if(counterBottom+counterTop== board.getWinningNumber() - 1) return 100;
+    return counterBottom+counterTop;
+}
+
+int AI::evaluateHorizontally(const Board &board, int xCoordinate, int yCoordinate, char currentPlayer) const
+{
+    int start =0, end = board.getSize() -1;
+    if(xCoordinate - board.getWinningNumber() +1 >= 0 ) start = xCoordinate -  board.getWinningNumber() +1;
+    if(xCoordinate +  board.getWinningNumber() - 1 <= end) end = xCoordinate + board.getWinningNumber() - 1;
+
+    int counterLeft = 0, counterRight = 0;
+    int opponentCounterLeft = 0, opponentCounterRight = 0;
+
+    for(int i = start; i < xCoordinate; ++i)
+    {
+        if (board.getFields()[i][yCoordinate]==currentPlayer)
+        {
+            counterLeft++;
+            opponentCounterLeft = 0;
+        }
+        else if (board.getFields()[i][yCoordinate]== ' ');
+        else
+        {
+            opponentCounterLeft++;
+            counterLeft = 0;
+        }
+    }
+
+    for(int i = end; i > xCoordinate; --i)
+    {
+        if (board.getFields()[i][yCoordinate]==currentPlayer)
+        {
+            counterRight++;
+            opponentCounterRight = 0;
+        }
+        else if (board.getFields()[i][yCoordinate]== ' ');
```

Comment:
There is a practise about https://en.wikipedia.org/wiki/Yoda_conditions especially in C like code. There are some users which recommends this, some of them no, I always use that :)